src/git.zsh

```diff
@@ -88,47 +86,92 @@ _fzf_complete_git() {
+                        completing_option_source=prefix
+                        break
+                    fi
+                    ;;
+            esac
+
+            if [[ $current != -[A-Za-z][A-Za-z]* ]]; then
+                break
+            fi
 
-        local git_options_commit_message_completion=(m message)
-        if _fzf_complete_git_has_options "$last_argument" "$prefix" $git_options_commit_message_completion; then
-            prefix_option=$(_fzf_complete_git_option_prefix) _fzf_complete_git-commit-messages '' $@
-            return
-        fi
+            current=${current/-[A-Za-z]/-}
+        done
 
-        local git_options_nothing_completion=(author date)
-        if _fzf_complete_git_has_options "$last_argument" "$prefix" $git_options_nothing_completion; then
-            return
-        fi
+        current=$last_argument
+        while [[ $current != '' ]]; do
+            if [[ ${git_options_argument_required[(r)$current]} != '' ]]; then
+                completing_option=$current
+                completing_option_source=last_argument
+                break
+            fi
 
-        local git_options_file_completion=(F t file pathspec-from-file template)
-        if _fzf_complete_git_has_options "$last_argument" "$prefix" $git_options_file_completion; then
-            _fzf_path_completion "${prefix/--*=}" $@$(_fzf_complete_git_option_prefix)
-            return
-        fi
+            if [[ $current != -[A-Za-z][A-Za-z]* ]]; then
+                break
+            fi
 
-        local git_options_cleanup_mode_completion=(cleanup)
-        if _fzf_complete_git_has_options "$last_argument" "$prefix" $git_options_cleanup_mode_completion; then
-            _fzf_complete '' $@ < <(awk -v prefix=$(_fzf_complete_git_option_prefix) '{ print prefix $0 }' <<< ${(F)cleanup_modes})
-            return
-        fi
+            current=${current/-[A-Za-z]/-}
+        done
 
-        local git_options_untracked_files_mode_completion=(u untracked-files)
-        if _fzf_complete_git_has_options "$last_argument" "$prefix" $git_options_untracked_files_mode_completion; then
-            _fzf_complete '' $@ < <(awk -v prefix=$(_fzf_complete_git_option_prefix) '{ print prefix $0 }' <<< ${(F)untracked_file_modes})
-            return
+        if [[ $completing_option_source = prefix ]]; then
+            prefix_option=$prefix
         fi
 
-        _fzf_complete_git-unstaged-files '--untracked-files=no' "--multi $_fzf_complete_preview_git_diff $FZF_DEFAULT_OPTS" $@
+        case $completing_option in
+            -c|-C|--fixup|--reedit-message|--reuse-message|--squash)
+                _fzf_complete_git-commits '' $@
+                ;;
+
+            -m|--message)
+                _fzf_complete_git-commit-messages '' $@
+                ;;
+
+            --author|--date)
+                ;;
+
+            -F|-t|--file|--pathspec-from-file|--template)
+                _fzf_path_completion "${prefix/--*=}" $@$prefix_option
```

Comment:
`-Ffoo**` isn't behave as expected if the second argument of `_fzf_path_completion` is `${prefix/--*=}`. Can you change it as below?
https://github.com/chitoku-k/fzf-zsh-completions/blob/9529d750e763ef8e32501ae27faeb9183aeacb4a/src/git.zsh#L113

Reply:
`$num` is the position where short option that takes a value appears.

Reply:
I don't want `_fzf_complete_git_option_prefix` to revive as I removed it. I think I need to add some support anyway.

Reply:
Sorry. It is not second but first.

Reply:
I fixed it. Can you check whether it works?

Reply:
Looks good 💪